public/app/core/services/util_srv.ts

```diff
@@ -15,8 +20,27 @@ export class UtilSrv {
     appEvents.on(CoreEvents.showModal, this.showModal.bind(this), this.$rootScope);
     appEvents.on(CoreEvents.hideModal, this.hideModal.bind(this), this.$rootScope);
     appEvents.on(CoreEvents.showConfirmModal, this.showConfirmModal.bind(this), this.$rootScope);
+    appEvents.on(CoreEvents.showModalReact, this.showModalReact.bind(this), this.$rootScope);
   }
 
+  showModalReact(options: any) {
```

Comment:
I need this: https://github.com/grafana/grafana/pull/22395#discussion_r384372501 :)

Reply:
Yea, that was tricky. I also got some strange theme hadnling until I wraped component with `provideTheme()`. I'm not sure about wrapping with `WrapInProvider()`, because right now I don't need any state from redux store.